src/crewai/tools/agent_tools/add_image_tool.py

```diff
@@ -0,0 +1,40 @@
+from crewai.tools.base_tool import BaseTool
+from pydantic import BaseModel, Field
+
+
+class AddImageToolSchema(BaseModel):
+    image_url: str = Field(..., description="The URL or path of the image to add")
+    action: str = Field(
+        default="Please provide a detailed description of this image, including all visual elements, context, and any notable details you can observe.",
+        description="Optional context or question about the image"
+    )
+
+
+class AddImageTool(BaseTool):
+    """Tool for adding images to the content"""
+
+    name: str = "Add image to content"
+    description: str = "See image to understand it's content, you can optionally ask a question about the image"
+    args_schema: type[BaseModel] = AddImageToolSchema
+
+    def _run(
+        self,
+        image_url: str,
+        action: str = None,
+        **kwargs,
+    ) -> dict:
+        action = action or "Please provide a detailed description of this image, including all visual elements, context, and any notable details you can observe."
+        content = [
+            {"type": "text", "text": action},
+            {
+                "type": "image_url",
+                "image_url": {
+                    "url": image_url,
```

Comment:
How will this work for local images? Asking cause the OpenAI Vision has a base64 strategy to handle such cases.

Reply:
depends on the model, for most you need to have uploaded image or base64, but that is out of our control